pkg/operator/mon/spec.go

```diff
@@ -91,7 +91,7 @@ func (c *Cluster) makeMonPod(config *monConfig, nodeName string) *v1.Pod {
 	podSpec := v1.PodSpec{
 		Containers:    []v1.Container{container},
 		RestartPolicy: v1.RestartPolicyAlways,
-		NodeSelector:  map[string]string{apis.LabelHostname: nodeName},
+		NodeSelector:  map[string]string{apis.LabelHostname: nodeId},
```

Comment:
if nodeID is a name instead of a hostname, then we may need to use different key here instead of `LabelHostName`.

Reply:
If name is used as a node identifier then there is no way I can think of to make a node discrimination in the mon replicasets. Nodeselectors can only be labels https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels 

What do you think that would be the best solution, then? Delete the nodeselector?

Reply:
@jbw976 ? :)

Reply:
Curious to hear if @jbw976 has any opinions on @dgonzalezruiz's question. I've also ran into issue #1264 and it unfortunately has seemed to be a total blocker for getting rook up and running properly..

Reply:
From all the environments I've seen, the `kubernetes.io/hostname` label has always been set on the nodes, regardless of if it matches the node name field or not.  Your environment has this label set also.  I suspect that since the `getNodeId` function has a preference for the hostname field, which comes from the `kubernetes.io/hostname` label, that this selector will work as intended.